packages/devextreme-cli/utility/package-manager.js

```diff
@@ -0,0 +1,65 @@
+const fs = require('fs');
+const path = require('path');
+const runCommand = require('./run-command');
+const yarnLockfile = require('@yarnpkg/lockfile');
+
+const getLockFileName = (cwd) => {
+    return isYarn(cwd) ? 'yarn.lock' : 'package-lock.json';
+};
+
+const getPackageManager = (cwd) => {
+    return isYarn(cwd) ? 'yarn' : 'npm';
+};
+
+const getLockFile = (cwd) => {
+    const lockFilePath = path.join(cwd, getLockFileName(cwd));
+    if(!fs.existsSync(lockFilePath)) {
+        return;
+    }
+
+    return isYarn(cwd) ? getYarnLock(lockFilePath) : getPackageLock(lockFilePath);
+};
+
+const getPackageLock = (packageLockPath) => {
+    return require(packageLockPath);
+};
+
+const getYarnLock = (yarnLockPath) => {
+    return yarnLockfile.parse(fs.readFileSync(yarnLockPath, 'utf8'));
+};
+
+const installPackage = (packageName, evaluatingOptions, options) => {
+    const cwd = evaluatingOptions && evaluatingOptions.cwd;
+    const instalationOptions = options && options[getPackageManager()];
+    const commandArguments = isYarn(cwd) ? ['add'] : ['install'];
+
+    if(instalationOptions) {
+        commandArguments.concat(instalationOptions);
+    }
+
+    commandArguments.push(packageName);
+
+    return runCommand(getPackageManager(cwd), commandArguments, evaluatingOptions);
+};
+
+const isYarn = (cwd) => {
+    const lockFile = 'yarn.lock';
+    return cwd ? fs.existsSync(path.join(cwd, lockFile)) : fs.existsSync(lockFile);
+};
+
+const run = (commands, evaluatingOptions) => {
+    return runCommand(getPackageManager(), commands, evaluatingOptions);
+};
+
+const runInstall = (evaluatingOptions) => {
+    const cwd = evaluatingOptions && evaluatingOptions.cwd;
+    return runCommand(getPackageManager(cwd), ['install'], evaluatingOptions);
+};
+
+module.exports = {
+    getLockFile,
+    installPackage,
+    isYarn,
```

Comment:
Method `isYarn` breaks an incapsulation. Just add one more method which will be a getter for the dependencies object.
It's better to put isYarn logic inside 'getPackageManager' method and use 'getPackageManager' in other places instead of the `isYarn` method. It will allow you to replace all ternar operators with dictionaries usage. For example:
```
const packageManagerConfigs = {
  npm: {
    installCommand: 'install',
    getLockFile: getPackageLock,
    getDependencies: (obj) => obj.dependencies
  },
  yarn: {
    installCommand: 'add',
    getLockFile: getYarnLock ,
    getDependencies: (obj) => obj.object
  }
};
...
// And then this
isYarn(cwd) ? ['add'] : ['install'];
// becomes to be something like this
packageManagerConfigs[getPackageManager(cwd)].installCommand;
```
Now we don't have a magic strings in code (only in dictionaries). And now we can easily add any new exotic package manager in a future.

One more thing. It's not obvious why we put `cwd` somewhere and somewhere don't. It would be great if it'll be clear from code without additional explanation.